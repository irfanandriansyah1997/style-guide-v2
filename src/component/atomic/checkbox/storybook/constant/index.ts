import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_CHECKBOX_COMPONENT: IStorybookArgsItem[] = [
  {
    action: `clicked`,
    argsName: `onChangeSelection`,
    description: `This method will be invoked when user try to click checkbox component`,
    table: {
      category: `Event`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `(value: IDefaultText) => void`
      }
    }
  },
  {
    argsName: `children`,
    description: `Label checkbox component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ReactNode`
      }
    }
  },
  {
    argsName: `active`,
    control: `boolean`,
    description: `the purpose this props for check this component is active or otherwise`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean | undefined`
      }
    }
  },
  {
    argsName: `className`,
    description: `Classname list for customize classname each html tags on checkbox component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `IToggleClassnameList`
      }
    }
  },
  {
    argsName: `value`,
    control: false,
    description: `Value contains on checkbox componentValue checkbox component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `IDefaultText`
      }
    }
  },
  {
    argsName: `labelTag`,
    control: {
      options: [`p`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `span`],
      type: `select`
    },
    description: `Element tag will be rendered on label component`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `ITextType`
      }
    }
  }
];
