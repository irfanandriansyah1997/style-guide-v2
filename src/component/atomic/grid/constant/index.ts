import { Property } from 'csstype';

import { IGridSize } from '@/atomic/grid/interface';

export const DEFAULT_GRID_ALIGN_ITEM: Property.AlignItems[] = [
  `-moz-initial`,
  `baseline`,
  `center`,
  `end`,
  `flex-end`,
  `flex-start`,
  `inherit`,
  `initial`,
  `normal`,
  `revert`,
  `self-end`,
  `self-start`,
  `start`,
  `stretch`,
  `unset`
];

export const DEFAULT_GRID_JUSTIFY_CONTENT: Property.JustifyContent[] = [
  `-moz-initial`,
  `center`,
  `end`,
  `flex-end`,
  `flex-start`,
  `inherit`,
  `initial`,
  `left`,
  `normal`,
  `revert`,
  `right`,
  `space-around`,
  `space-between`,
  `space-evenly`,
  `start`,
  `stretch`,
  `unset`
];

export const DEFAULT_GRID_SIZE: IGridSize[] = [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
  `auto`
];
