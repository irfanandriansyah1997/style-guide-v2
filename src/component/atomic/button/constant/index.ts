import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_BUTTON_COMPONENT: IStorybookArgsItem[] = [
  {
    action: `clicked`,
    argsName: `onClick`
  },
  {
    argsName: `children`,
    control: false,
    description: `Button component text`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ReactNode`
      }
    }
  },
  {
    argsName: `outline`,
    description: `Set outline style of button`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rounded`,
    description: `Set rounded type of button`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rtl`,
    description: `Set right to left button content`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rtl`,
    description: `Set right to left button content`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `size`,
    control: {
      options: [`big`, `default`, `small`],
      type: `inline-radio`
    },
    description: `Set button size, eg. \`big\` \`default\` \`small\``,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `default`
      },
      type: {
        summary: `IButtonSize`
      }
    }
  },
  {
    argsName: `size`,
    control: {
      options: [`big`, `default`, `small`],
      type: `inline-radio`
    },
    description: `Set button size, eg. \`big\` \`default\` \`small\``,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `default`
      },
      type: {
        summary: `IButtonSize`
      }
    }
  },
  {
    argsName: `theme`,
    control: {
      options: [`light`, `dark`],
      type: `inline-radio`
    },
    description: `Set button theme`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `light`
      },
      type: {
        summary: `IButtonThemeOption`
      }
    }
  },
  {
    argsName: `className`,
    description: `Classname list for customize classname each html tags on button component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `IButtonClassnameList`
      }
    }
  },
  {
    argsName: `icon`,
    control: false,
    description: `Set the icon component of button`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ReactNode`
      }
    }
  }
];
