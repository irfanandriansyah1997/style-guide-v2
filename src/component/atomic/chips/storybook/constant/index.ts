import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_CHIPS_COMPONENT: IStorybookArgsItem[] = [
  {
    action: `clicked`,
    argsName: `on`,
    description: `This method will be invoked when user try to interact chips component`,
    table: {
      category: `Event`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `MouseEventHandler<HTMLButtonElement>`
      }
    }
  },
  {
    argsName: `label`,
    control: {
      type: `text`
    },
    description: `chips label text`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `string`
      }
    },
    type: {
      required: true
    }
  },
  {
    argsName: `outline`,
    description: `Set outline style of tags`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `className`,
    description: `Classname list for customize classname each html tags on tags component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `IChipsClassnameList`
      }
    }
  }
];
