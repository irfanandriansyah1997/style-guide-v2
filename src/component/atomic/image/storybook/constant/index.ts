import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_IMAGE_COMPONENT: IStorybookArgsItem[] = [
  {
    argsName: `alt`,
    description: `Attribute for define alternative text on images tag`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `src`,
    description: `Attribute specifies the path (URL) to the image`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    },
    type: {
      required: true
    }
  },
  {
    argsName: `className`,
    control: {
      type: `text`
    },
    description: `Attribute for customize classname on img tags`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `height`,
    control: {
      max: 400,
      min: 0,
      type: `range`
    },
    defaultValue: `-`,
    description: `Attribute for customize height image`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number | string`
      }
    }
  },
  {
    argsName: `width`,
    control: {
      max: 400,
      min: 0,
      type: `range`
    },
    description: `Attribute for customize width image`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number | string`
      }
    }
  },
  {
    argsName: `objectFit`,
    control: {
      options: [
        `-moz-initial`,
        `inherit`,
        `initial`,
        `revert`,
        `unset`,
        `contain`,
        `cover`,
        `fill`,
        `none`,
        `scale-down`
      ],
      type: `select`
    },
    description: `CSS property sets how the content should be resized to fits its container`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `initial`
      },
      type: {
        summary: `Property.ObjectFit`
      }
    }
  },
  {
    argsName: `shape`,
    control: {
      options: [
        `circle`,
        `square`,
        `rounded`,
        `rounded-bottom`,
        `rounded-left`,
        `rounded-top`,
        `rounded-right`
      ],
      type: `select`
    },
    description: `Attribute for define image shape, such as circle, rounded etc`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `initial`
      },
      type: {
        summary: `Property.ObjectFit`
      }
    }
  }
];

export const CUSTOM_ARGS_IMAGE_RESPONSIVE_COMPONENT = [
  ...CUSTOM_ARGS_IMAGE_COMPONENT.filter((item) => item.argsName !== `src`),
  {
    argsName: `sizes`,
    control: {
      type: `text`
    },
    defaultValue: `-`,
    description: `Attribute for tell the browser to assume the image fills the entire browser width`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    }
  }
];
