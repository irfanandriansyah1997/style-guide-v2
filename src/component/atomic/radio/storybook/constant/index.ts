import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_RADIO_COMPONENT: IStorybookArgsItem[] = [
  {
    action: `clicked`,
    argsName: `on`,
    description: `This method will be invoked when user interact radio component`,
    table: {
      category: `Event`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `(param: IEventOnChangeSelection) => void`
      }
    }
  },
  {
    argsName: `children`,
    description: `Label radio component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ReactNode`
      }
    }
  },
  {
    argsName: `active`,
    control: `boolean`,
    description: `the purpose this props for check this component is active or otherwise`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean | undefined`
      }
    }
  },
  {
    argsName: `className`,
    description: `Classname list for customize classname each html tags on radio component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `IToggleClassnameList`
      }
    }
  },
  {
    argsName: `value`,
    control: false,
    description: `Value contains on radio componentValue radio component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `IDefaultText`
      }
    }
  },
  {
    argsName: `labelTag`,
    control: {
      options: [`p`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `span`],
      type: `select`
    },
    description: `Element tag will be rendered on label component`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `ITextType`
      }
    }
  }
];
