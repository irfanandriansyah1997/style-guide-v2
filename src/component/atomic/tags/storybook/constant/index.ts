import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_TAGS_COMPONENT: IStorybookArgsItem[] = [
  {
    action: `clicked`,
    argsName: `onClick`
  },
  {
    argsName: `children`,
    control: false,
    description: `tags component text`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ReactNode`
      }
    }
  },
  {
    argsName: `outline`,
    description: `Set outline style of tags`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `inverted`,
    description: `Set inverted on background color of tags`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rounded`,
    description: `Set rounded type of tags`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rtl`,
    description: `Set right to left tags content`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `rtl`,
    description: `Set right to left tags content`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `false`
      },
      type: {
        summary: `boolean`
      }
    }
  },
  {
    argsName: `size`,
    control: {
      options: [`default`, `small`],
      type: `inline-radio`
    },
    description: `Set tags size, eg. \`default\` \`small\``,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `default`
      },
      type: {
        summary: `ITagsSize`
      }
    }
  },
  {
    argsName: `theme`,
    control: {
      options: [`light`, `dark`],
      type: `inline-radio`
    },
    description: `Set tags theme`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `light`
      },
      type: {
        summary: `ITagsThemeOption`
      }
    }
  },
  {
    argsName: `className`,
    description: `Classname list for customize classname each html tags on tags component`,
    table: {
      category: `Content`,
      defaultValue: {
        summary: `undefined`
      },
      type: {
        summary: `ITagsClassnameList`
      }
    }
  }
];
