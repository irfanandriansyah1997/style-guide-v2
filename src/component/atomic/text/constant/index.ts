import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_TEXT_COMPONENT: IStorybookArgsItem[] = [
  {
    argsName: `children`,
    control: {
      type: `text`
    },
    defaultValue: ``,
    description: `Content on text component`,
    table: {
      category: `Content`,
      type: {
        summary: `ReactNode`
      }
    },
    type: {
      required: true
    }
  },
  {
    argsName: `color`,
    control: {
      type: `color`
    },
    description: `What text color to use`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `fontFamily`,
    control: {
      type: `text`
    },
    defaultValue: `'Museo Sans', sans-serif`,
    description: `What typeface to use`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `fontSize`,
    control: {
      max: 64,
      min: 12,
      type: `range`
    },
    defaultValue: `-`,
    description: `CSS property sets the size of the font`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `fontWeight`,
    control: {
      options: [100, 200, 300, 400, 500, 600, 700],
      type: `select`
    },
    defaultValue: 400,
    description: `CSS property sets the weight (or boldness) of the font`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `lineHeight`,
    control: {
      max: 64,
      min: 12,
      type: `range`
    },
    defaultValue: `-`,
    description: `CSS property sets the height of a line box`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `textAlign`,
    control: {
      options: [
        `-moz-initial`,
        `inherit`,
        `initial`,
        `revert`,
        `unset`,
        `center`,
        `end`,
        `justify`,
        `left`,
        `match-parent`,
        `right`,
        `start`
      ],
      type: `select`
    },
    description: `CSS property sets the horizontal alignment of a block element`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `Property.TextAlign`
      }
    }
  },
  {
    argsName: `textDecoration`,
    control: {
      options: [
        `auto`,
        `blink`,
        `dashed`,
        `dotted`,
        `double`,
        `from-font`,
        `grammar-error`,
        `line-through`,
        `none`,
        `overline`,
        `solid`,
        `spelling-error`,
        `underline`,
        `wavy`
      ],
      type: `select`
    },
    description: `CSS property sets the color of decorations added to text by \`text-decoration-line\``,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `Property.TextDecoration`
      }
    }
  }
];
