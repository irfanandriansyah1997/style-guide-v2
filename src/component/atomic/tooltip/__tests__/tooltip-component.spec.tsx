import { fireEvent, render } from '@testing-library/react';

import Tooltip from '@/atomic/tooltip';
import { STYLE_GUIDE_COLOR } from '@/constant/color';

describe(`Test Tooltip component`, () => {
  it(`Simulate Hover Tooltip`, () => {
    const { container } = render(
      <Tooltip
        content="Sample Content"
        position="right"
        background={STYLE_GUIDE_COLOR.light100}
        className={{
          toggle: `sample-class-toggle`,
          tooltip: `sample-class-tooltip`
        }}
      >
        <h1>Hover This Heading</h1>
      </Tooltip>
    );

    expect(container.outerHTML).toMatchSnapshot();

    if (container.firstChild) {
      const {
        childNodes: firstChildNodes
      } = container.firstChild as HTMLElement;
      const toggle = firstChildNodes[0] as HTMLElement;

      expect((container.firstChild as HTMLElement).childElementCount).toBe(1);
      expect(toggle).toHaveClass(`sample-class-toggle`);
      expect(toggle).toHaveTextContent(`Hover This Heading`);
      expect((toggle.firstChild as HTMLElement).tagName).toBe(`H1`);

      fireEvent.mouseEnter(toggle);
      expect(container.outerHTML).toMatchSnapshot();

      const {
        childNodes: secondChildNodes
      } = container.firstChild as HTMLElement;
      expect((container.firstChild as HTMLElement).childElementCount).toBe(2);
      const tooltip = secondChildNodes[1] as HTMLElement;
      const triangle = tooltip.childNodes[0];

      expect(tooltip.childElementCount).toBe(1);
      expect(tooltip).toHaveClass(`a-tooltip__content`);
      expect(tooltip).toHaveClass(`a-tooltip__content--position-right`);
      expect(tooltip).toHaveClass(`sample-class-tooltip`);
      expect(tooltip).toHaveStyle({
        backgroundColor: `rgb(250, 250, 250)`
      });
      expect(tooltip).toHaveTextContent(`Sample Content`);

      expect(triangle).toHaveClass(`a-tooltip__triangle`);
      expect(triangle).toHaveStyle({ borderRightColor: `#fafafa` });
    }
  });

  it(`Simulate Hover Tooltip Bottom`, () => {
    const { container } = render(
      <Tooltip
        content="Sample Content"
        position="bottom"
        background={STYLE_GUIDE_COLOR.danger500}
        className={{
          toggle: `sample-class-toggle`,
          tooltip: `sample-class-tooltip`
        }}
      >
        <h1>Hover This Heading</h1>
      </Tooltip>
    );

    expect(container.outerHTML).toMatchSnapshot();

    if (container.firstChild) {
      const {
        childNodes: firstChildNodes
      } = container.firstChild as HTMLElement;
      const toggle = firstChildNodes[0] as HTMLElement;

      expect((container.firstChild as HTMLElement).childElementCount).toBe(1);
      expect(toggle).toHaveClass(`sample-class-toggle`);
      expect(toggle).toHaveTextContent(`Hover This Heading`);
      expect((toggle.firstChild as HTMLElement).tagName).toBe(`H1`);

      fireEvent.mouseEnter(toggle);
      expect(container.outerHTML).toMatchSnapshot();

      const {
        childNodes: secondChildNodes
      } = container.firstChild as HTMLElement;
      expect((container.firstChild as HTMLElement).childElementCount).toBe(2);
      const tooltip = secondChildNodes[1] as HTMLElement;
      const triangle = tooltip.childNodes[0];

      expect(tooltip.childElementCount).toBe(1);
      expect(tooltip).toHaveClass(`a-tooltip__content`);
      expect(tooltip).toHaveClass(`a-tooltip__content--position-bottom`);
      expect(tooltip).toHaveClass(`sample-class-tooltip`);
      expect(tooltip).toHaveStyle({
        backgroundColor: `rgb(255, 79, 56)`
      });
      expect(tooltip).toHaveTextContent(`Sample Content`);

      expect(triangle).toHaveClass(`a-tooltip__triangle`);
      expect(triangle).toHaveStyle({ borderBottomColor: `#ff4f38` });
    }
  });

  it(`Simulate Hover Tooltip Not Visible`, () => {
    const { container } = render(
      <Tooltip
        content="Sample Content"
        position="right"
        visible={false}
        background={STYLE_GUIDE_COLOR.light100}
        className={{
          toggle: `sample-class-toggle`,
          tooltip: `sample-class-tooltip`
        }}
      >
        <h1>Hover This Heading</h1>
      </Tooltip>
    );

    expect(container.outerHTML).toMatchSnapshot();

    if (container.firstChild) {
      const {
        childNodes: firstChildNodes
      } = container.firstChild as HTMLElement;
      const toggle = firstChildNodes[0] as HTMLElement;

      expect((container.firstChild as HTMLElement).childElementCount).toBe(1);
      expect(toggle).toHaveClass(`sample-class-toggle`);
      expect(toggle).toHaveTextContent(`Hover This Heading`);
      expect((toggle.firstChild as HTMLElement).tagName).toBe(`H1`);

      fireEvent.mouseEnter(toggle);
      expect(container.outerHTML).toMatchSnapshot();

      expect((container.firstChild as HTMLElement).childElementCount).toBe(1);
    }
  });
});
