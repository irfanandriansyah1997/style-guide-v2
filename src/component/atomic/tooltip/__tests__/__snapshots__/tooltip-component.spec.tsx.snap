// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component Simulate Hover Tooltip 1`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div></div></div>"`;

exports[`Test Tooltip component Simulate Hover Tooltip 2`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div><div class=\\"absolute block fade a-tooltip__content a-tooltip__content--position-right fade-entering sample-class-tooltip\\" style=\\"background: rgb(250, 250, 250);\\"><div class=\\"absolute a-tooltip__triangle\\" style=\\"border-right-color: #fafafa;\\"></div>Sample Content</div></div></div>"`;

exports[`Test Tooltip component Simulate Hover Tooltip Bottom 1`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div></div></div>"`;

exports[`Test Tooltip component Simulate Hover Tooltip Bottom 2`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div><div class=\\"absolute block fade a-tooltip__content a-tooltip__content--position-bottom fade-entering sample-class-tooltip\\" style=\\"background: rgb(255, 79, 56);\\"><div class=\\"absolute a-tooltip__triangle\\" style=\\"border-bottom-color: #ff4f38;\\"></div>Sample Content</div></div></div>"`;

exports[`Test Tooltip component Simulate Hover Tooltip Not Visible 1`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div></div></div>"`;

exports[`Test Tooltip component Simulate Hover Tooltip Not Visible 2`] = `"<div><div class=\\"relative inline-block\\"><div class=\\"sample-class-toggle\\"><h1>Hover This Heading</h1></div></div></div>"`;
